src/librustc/middle/resolve_lifetime.rs

```diff
@@ -992,11 +992,14 @@ impl<'a, 'tcx> LifetimeContext<'a, 'tcx> {
             let unsubst = if let Some(id) = self.hir_map.as_local_node_id(def_id) {
                 &map.object_lifetime_defaults[&id]
             } else {
-                let cstore = &self.sess.cstore;
                 self.xcrate_object_lifetime_defaults.entry(def_id).or_insert_with(|| {
-                    cstore.item_generics_cloned(def_id).types.into_iter().map(|def| {
-                        def.object_lifetime_default
-                    }).collect()
+                    ty::tls::with_opt(|opt_tcx| {
```

Comment:
The reason `item_generics_cloned` is named the way it is (I should've commented it, shouldn't I?) is because it's `item_generics` (what is now `generics_of`), but returning a *clone* of the data that does not require a `TyCtxt`.